### PR TITLE
server: error logging

### PR DESCRIPTION
**Pull request by kingsleyzissou**
Currently we are only logging and recording metrics for internal server errors
with exact code 500. This commit now ensures that 5xx errors are logged
and the metrics recorded.

### CODE REVIEW

File: internal/v1/server.go [hunk 1]
```diff
@@ -413,13 +413,13 @@ func (h *Handlers) ComposeImage(ctx echo.Context) error {
...
 	}
 
-	quotaOk, err := common.CheckQuota(idHeader.Identity.AccountNumber, h.server.db, h.server.quotaFile)
-	if err != nil {
-		return err
-	}
-	if !quotaOk {
-		return echo.NewHTTPError(http.StatusForbidden, "Quota exceeded for user")
-	}
+	// quotaOk, err := common.CheckQuota(idHeader.Identity.AccountNumber, h.server.db, h.server.quotaFile)
```
Comment (ondrejbudai) [inline internal/v1/server.go:416]: I think this is not intentional, right?

**Review by ondrejbudai — CHANGES_REQUESTED** [review]

File: internal/v1/server.go (hunk 1 above)
Comment (kingsleyzissou) [reply internal/v1/server.go:416]: Definitely not! Had to comment that out for it to run locally. Will fix that now

**Comment by kingsleyzissou** [comment]
@ondrejbudai are we still having issues with the CI?

**Comment by ondrejbudai** [comment]
Yeah, sorry, I need to figure out what to do with it if we cannot merge https://github.com/osbuild/image-builder/pull/274 for a long time.

**Comment by ondrejbudai** [comment]
Needs a rebase, that will fix the CI.

File: internal/v1/server.go [hunk 2]
```diff
@@ -480,9 +480,11 @@ func (h *Handlers) ComposeImage(ctx echo.Context) error {
 
 	err = h.server.db.InsertCompose(composeResult.Id, idHeader.Identity.AccountNumber, idHeader.Identity.Internal.OrgId, rawCR)
 	if err != nil {
+		h.server.logger.Info("Error inserting id into db", err)
```
Comment (Gundersanne) [inline internal/v1/server.go:483]: this can be a `.Error` level alert, useful when filtering messages by level in kibana

**Review by teg — APPROVED** [review]

**Review by ondrejbudai — APPROVED** [review]

**Pull request merged**